routes/test/signupRoutes.spec.js

```diff
@@ -0,0 +1,155 @@
+const request = require('supertest');
+const express = require('express');
+const router = require('../signupRoutes'); // Import your router file here
+const connectDb = require('../../__tests__/fixtures/db')
+const makeFakeUser = require('../../__tests__/fixtures/fakeUser')
+const mongoose = require('mongoose');
+
+
+const app = express();
+app.use(express.json());
+app.use('/api/signup', router); // Mount the router under '/api/signup' path
+
+// Start the Express app on a specific port for testing
+const PORT = 5000; // Choose a port for testing
+const server = app.listen(PORT, () => {
+  console.log(`Express server is running on port ${PORT}`);
+});
+
+// Create a fake user
+let fakeUser = makeFakeUser();
+
+
+describe('Signup User route', () => {
+
+  let db; // Store the database connection
+
+  beforeAll(async () => {
+    db = await connectDb(); // Connect to the database
+
+    // Insert the fake user into the database
+    //await db.collection('users').insertOne(fakeUser);
+  });
+
+  it('Check that the endpoint saves the user in the database', async () => {
+    const input = {
+      name: "test user",
+      email: fakeUser.email,
+      password: "ABC123456!",
+      confirmPassword: "ABC123456!",
```

Comment:
confirmPassword is only used in the frontend, it's not being sent to the backend.